### PR TITLE
Match table for db

### PR DESCRIPTION
**Pull request by diego-centella-jalafund**

### CODE REVIEW

File: src/Program.cs [hunk 1]
```diff
@@ -1,8 +1,14 @@
 using MatchService.Infrastructure;
 using MatchService.RequestPipeline;
-
+using Npgsql; // Para NpgsqlConnection
+using System.Data; 
+using MatchService.Domain;
+using MatchService.Data.Concretes;
 var builder = WebApplication.CreateBuilder(args);
 
+builder.Services.AddSingleton<IDbConnection>((sp) =>
```
Comment (luis-espinoza-jalafund) [inline src/Program.cs]: It is not necessary to add this, the table is already created with the script without using Services.AddSingleton.

File: src/Program.cs [hunk 2]
```diff
@@ -1,8 +1,14 @@
 using MatchService.Infrastructure;
 using MatchService.RequestPipeline;
-
+using Npgsql; // Para NpgsqlConnection
```
Comment (luis-espinoza-jalafund) [inline src/Program.cs]: Do not leave comments without utility

File: src/Data/Scripts/Match.sql [hunk 3]
```diff
@@ -0,0 +1,7 @@
+CREATE TABLE IF NOT EXISTS match (
+    id SERIAL PRIMARY KEY,
+    "user" VARCHAR(255) NOT NULL,
+    worker VARCHAR(255) NOT NULL,
+    skill VARCHAR(255) NOT NULL,
+    created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP
+);
```
Comment (luis-espinoza-jalafund) [inline src/Data/Scripts/Match.sql]: Missing endline

File: src/Domain/Match.cs [hunk 4]
```diff
@@ -0,0 +1,9 @@
+namespace MatchService.Domain;
+
+public class Match
+{
+    public int Id { get; set; }
+    public required string User { get; set; }
+    public required string Worker { get; set; }
+    public required string skill { get; set; } 
+}
```
Comment (luis-espinoza-jalafund) [inline src/Domain/Match.cs]: Missing endline

File: src/Data/Scripts/Match.sql [hunk 5]
```diff
@@ -0,0 +1,7 @@
+CREATE TABLE IF NOT EXISTS match (
+    id SERIAL PRIMARY KEY,
```
Comment (luis-espinoza-jalafund) [inline src/Data/Scripts/Match.sql]: The ID should be a GUIID

File: src/Domain/Match.cs [hunk 6]
```diff
@@ -0,0 +1,9 @@
+namespace MatchService.Domain;
+
+public class Match
+{
+    public int Id { get; set; }
```
Comment (luis-espinoza-jalafund) [inline src/Domain/Match.cs]: The ID should be a GUID not an int

**Review by luis-espinoza-jalafund — CHANGES_REQUESTED** [review]

**Comment by luis-espinoza-jalafund** [comment]
You must also add the JSON that we will use for the User and Workers in order to make the connections between these tables and apply the match.

File: src/Data/Scripts/Match.sql [hunk 7]
```diff
@@ -0,0 +1,7 @@
...
+    "user" VARCHAR(255) NOT NULL,
+    worker VARCHAR(255) NOT NULL,
+    skill VARCHAR(255) NOT NULL,
```
Comment (luis-espinoza-jalafund) [inline src/Data/Scripts/Match.sql]: the skills will be text only? what does the other group say about this?

File: src/Domain/Match.cs [hunk 8]
```diff
@@ -0,0 +1,9 @@
...
+    public required string User { get; set; }
+    public required string Worker { get; set; }
```
Comment (luis-espinoza-jalafund) [inline src/Domain/Match.cs]: is this the id of the user and worker? if so, why don't they have “ID” in the variable name?

File: src/Data/Scripts/Match.sql [hunk 9]
```diff
@@ -0,0 +1,7 @@
+CREATE TABLE IF NOT EXISTS match (
+    id UUID PRIMARY KEY DEFAULT uuid_generate_v4(),
```
Comment (luis-espinoza-jalafund) [inline src/Data/Scripts/Match.sql]: Here you use a postgres extension, did you make sure it is installed automatically for everyone?

File: src/Program.cs [hunk 10]
```diff
@@ -1,8 +1,12 @@
 using MatchService.Infrastructure;
 using MatchService.RequestPipeline;
-
+using Npgsql; 
+using System.Data; 
+using MatchService.Domain;
+using MatchService.Data.Concretes;
 var builder = WebApplication.CreateBuilder(args);
 
+builder.Services.AddTransient<DbInitializer>();
```
Comment (luis-espinoza-jalafund) [inline src/Program.cs]: This is also innecesary

File: src/Persistence/Jobs.json [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+      {
+        "id": "client-uuid-2",
+        "name": "Carlos Gómez",
+        "email_address": "carlos.gomez@example.com",
+        "category_id": "category-uuid-1"
+      }
+    ],
+    "worker": [
+      {
+        "id": "worker-uuid-1",
```
Comment (SebasthianSalpozz) [inline src/Persistence/Jobs.json]: Tus trabajadores no tienen ningun tipo de ubicacion para ir ver si estan en la zona del usuario

File: src/Persistence/Jobs.json [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+      },
+      {
+        "id": "category-uuid-2",
+        "rol": "worker"
+      }
+    ],
+    "user": [
+      {
+        "id": "client-uuid-1",
+        "name": "Juan Pérez",
```
Comment (SebasthianSalpozz) [inline src/Persistence/Jobs.json]: Tus usuarios no tienen ningun tipo de ubicacion para ir buscando alrededor

**Review by SebasthianSalpozz — CHANGES_REQUESTED** [review]

**Review by SebasthianSalpozz — COMMENTED** [review]
Talvez para manejar las ubicaciones podriamos usar la investigacion de postgis q esta en la wiki y asi poder realizar solo un query y todo esto que este seteado en una base de datos y solo realizar la consulta o como manejaremos las ubicaciones para el match ?

File: src/Data/Scripts/DataSQL.sql [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+    geom GEOMETRY(Point, 4326)
+);
+
+CREATE TABLE IF NOT EXISTS client (
+    id UUID PRIMARY KEY,
+    localization_id UUID NOT NULL,
+    name VARCHAR NOT NULL,
+    emailAddress VARCHAR NOT NULL,
+    categoryId UUID NOT NULL
+);
```
Comment (luis-espinoza-jalafund) [inline src/Data/Scripts/DataSQL.sql:31]: The work schedule column is missing, it is necessary to know the worker's availability.

File: src/Data/Scripts/DataSQL.sql [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+    user_worker_id UUID NOT NULL,
+    localization_id UUID NOT NULL,
+    created_at TIMESTAMP NOT NULL DEFAULT NOW(),
+    job_type VARCHAR(50) NOT NULL, 
+    status VARCHAR(20) NOT NULL CHECK (status IN ('not assigned', 'assigned', 'on the way', 'working', 'done')),
+    description TEXT,
+    price DECIMAL(10, 2),
+    CONSTRAINT fk_localization FOREIGN KEY (localization_id) REFERENCES localization(id) 
+);
+
```
Comment (luis-espinoza-jalafund) [inline src/Data/Scripts/DataSQL.sql:45]: the Speciality and WorkSchedule tables are missing

**Review by luis-espinoza-jalafund — CHANGES_REQUESTED** [review]

**Review by luis-espinoza-jalafund — APPROVED** [review]

File: src/Data/Scripts/DataSQL.sql [hunk 15]
```diff
@@ -0,0 +1,70 @@
...
+    (gen_random_uuid(), (SELECT id FROM localization WHERE name = 'Location B') ,'Jane Smith', 'jane@example.com', (SELECT id FROM category WHERE rol = 'user'), '550e8400-e29b-41d4-a716-446655440006'),
+    (gen_random_uuid(), (SELECT id FROM localization WHERE name = 'Location C') ,'Mike Brown', 'mike@example.com', (SELECT id FROM category WHERE rol = 'worker'), '550e8400-e29b-41d4-a716-446655440006');
+
+INSERT INTO jobs (id, user_client_id, user_worker_id, localization_id, created_at, job_type, status, description, price) VALUES
+    (gen_random_uuid(), (SELECT id FROM client WHERE name = 'John Doe'), (SELECT id FROM client WHERE name = 'Mike Brown'), (SELECT id FROM localization WHERE name = 'Location A'), NOW(), 'electrician', 'assigned', 'Job description for Job 1', 100.00),
+    (gen_random_uuid(), (SELECT id FROM client WHERE name = 'Jane Smith'), (SELECT id FROM client WHERE name = 'Mike Brown'), (SELECT id FROM localization WHERE name = 'Location B'), NOW(), 'plumber', 'working', 'Job description for Job 2', 200.00);
+
```
Comment (luis-espinoza-jalafund) [inline src/Data/Scripts/DataSQL.sql:70]: the project stopped compiling because of missing postgis dependencies, do a rebase and update your branch

**Review by luis-espinoza-jalafund — CHANGES_REQUESTED** [review]

**Review by luis-espinoza-jalafund — APPROVED** [review]

**Pull request closed without merging**